source/DSCResources/DSC_Computer/DSC_Computer.psm1

```diff
@@ -247,6 +255,30 @@ function Set-TargetResource
                     $addComputerParameters.Add("Server", $Server)
                 }
 
+                if (-not [System.String]::IsNullOrEmpty($Options)){
+                    <#
+                        See https://docs.microsoft.com/en-us/powershell/module/microsoft.powershell.management/add-computer?view=powershell-5.1#parameters for available options and their description
+                    #>
+                    $joinOptions = New-Object System.Collections.Generic.List[string]
+                    $available_options = @("AccountCreate","Win9XUpgrade","UnsecuredJoin","PasswordPass","JoinWithNewName","JoinReadOnly","InstallInvoke")
+                    foreach ($option in $Options) {
+                        if ($option -in $available_options) {
+                            if ($option -eq 'PasswordPass') {
+                                if ('UnsecuredJoin' -in $Options) {
+                                    # PasswordPass must be used in conjunction with UnsecuredJoin
+                                    if ([System.String]::IsNullOrEmpty($Credential.UserName)){
+                                        $joinOptions.Add($option)
+                                    }
+                                } else {
+                                    New-InvalidOperationException -Message "PasswordPass is valid only when teh UnsecuredJoin option is specified"
```

Comment:
I'm not 100% on how I should be using these error functions if you've got a quick doc on it.